### PR TITLE
Homework4: Andrey Kuksik

### PR DESCRIPTION
**Pull request by Kuxsoger**

### CODE REVIEW

File: src/homework4/decorator.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"""
+Создайте декоратор, который хранит в себе результаты вызовов функций
+и время получения результата за время запуска программы.
+"""
+
+import time
+
+
+def log_results(lst):
```
Comment (belovmd) [inline src/homework4/decorator.py]: It's a good idea but I want just to decorate function, not create other variables etc.

File: src/homework4/get_ranges.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+которая этот список “сворачивает”
+"""
+
+
+def get_ranges(lst):
+    result = ""
+    start, prev = lst[0], lst[0]
+    for num in lst[1:]:
+        if num - prev != 1:
+            result += str(start) + ('-' + str(prev)) * (prev != start) + ','
```
Comment (belovmd) [inline src/homework4/get_ranges.py]: here it's not the best solution as it looks too heavyweight

**Review by belovmd — COMMENTED** [review]
Thank you for your work please address these issues:

File: src/homework4/decorator.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+Создайте декоратор, который хранит в себе результаты вызовов функций
+и время получения результата за время запуска программы.
+"""
+
+import time
+
+
+def log_results(func):
+    if not hasattr(log_results, "logs"):
+        log_results.logs = []
```
Comment (belovmd) [inline src/homework4/decorator.py:11]: why can't we use logs = []?

**Review by belovmd — APPROVED** [review]
8

File: src/homework4/decorator.py (hunk 3 above)
Comment (belovmd) [reply src/homework4/decorator.py:11]: i will improve score if you can explain why do we need hasattr here. you can email explanation.

Comment (Kuxsoger) [reply src/homework4/decorator.py:11]: i sent a message to python.epam.q42019@gmail.com

**Comment by belovmd** [comment]
thank you. 9;

**Pull request closed without merging**